internal/resource-request-operator/offerUpdater.go

```diff
@@ -0,0 +1,191 @@
+	wait.Until(u.run, 2*time.Second, ctx.Done())
+}
+
+func (u *OfferUpdater) run() {
+	for u.processNextItem() {
+	}
+}
+
+func (u *OfferUpdater) processNextItem() bool {
+	obj, shutdown := u.queue.Get()
+	if shutdown {
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer u.queue.Done(obj)
+		var clusterID string
+		var ok bool
+		if clusterID, ok = obj.(string); !ok {
+			// As the item in the workqueue is actually invalid, we call
+			// Forget here else we'd go into a loop of attempting to
+			// process a work item that is invalid.
+			u.queue.Forget(obj)
+			return fmt.Errorf("error getting object %v from OfferUpater queue. It is not a string", obj)
+		}
+		if requeue, err := u.createOrUpdateOffer(clusterID); err != nil {
+			// Put the item back on the workqueue to handle any transient errors.
+			if requeue {
+				u.queue.AddRateLimited(clusterID)
+			}
+			return fmt.Errorf("error during updating ResourceOffer for cluster %s: %w", clusterID, err)
+		}
+		return nil
+	}(obj)
+	if err != nil {
+		klog.Errorf("Error occurred during ResourceOffer update %s", err)
+		return true
+	}
+	klog.Infof("Update cluster %s processed", obj.(string))
+
+	// requeue after timeout seconds
+	u.queue.AddAfter(obj, requeueTimeout+time.Duration(getRandomDelta())*time.Second)
+	return true
+}
+
+func (u *OfferUpdater) createOrUpdateOffer(clusterID string) (bool, error) {
+	list, err := u.getResourceRequest(clusterID)
+	if err != nil {
+		return true, err
+	} else if len(list.Items) != 1 {
+		// invalid clusterID
+		u.broadcasterInt.RemoveClusterID(clusterID)
+		return false, fmt.Errorf("ClusterID %s is no more valid. Deleting", clusterID)
+	}
+	request := list.Items[0]
+	resources := u.broadcasterInt.ReadResources(clusterID)
+	offer := &sharingv1alpha1.ResourceOffer{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: request.GetNamespace(),
+			Name:      offerPrefix + u.homeClusterID,
+		},
+	}
+
+	op, err := controllerutil.CreateOrUpdate(context.Background(), u.Client, offer, func() error {
+		if offer.ObjectMeta.CreationTimestamp.IsZero() {
+			offer.Labels = map[string]string{
+				discovery.ClusterIDLabel:         request.Spec.ClusterIdentity.ClusterID,
+				crdreplicator.LocalLabelSelector: "true",
+				crdreplicator.DestinationLabel:   request.Spec.ClusterIdentity.ClusterID,
+			}
+			spec := sharingv1alpha1.ResourceOfferSpec{
+				ClusterId: u.homeClusterID,
+				Images:    []corev1.ContainerImage{},
+				ResourceQuota: corev1.ResourceQuotaSpec{
+					Hard: resources,
+				},
+				Labels: u.broadcasterInt.GetConfig().Spec.DiscoveryConfig.ClusterLabels,
+			}
+			offer.Spec = spec
+			return controllerutil.SetControllerReference(&request, offer, u.scheme)
+		}
+		spec := sharingv1alpha1.ResourceOfferSpec{
+			ClusterId: u.homeClusterID,
+			Images:    []corev1.ContainerImage{},
+			ResourceQuota: corev1.ResourceQuotaSpec{
+				Hard: resources,
+			},
+			Labels: u.broadcasterInt.GetConfig().Spec.DiscoveryConfig.ClusterLabels,
+		}
```

Comment:
Doing so you would overwrite the entire spec. You should only change the fields you are controlling.